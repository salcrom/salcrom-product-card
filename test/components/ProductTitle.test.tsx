// import * as React from 'react';
import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';



describe('ProductTitle', () => {

    test('debe de mostrar el componente correctamente con el tÃ­tulo personalizado', () => {
        const wrapper = renderer.create(
            <ProductTitle title="Custom Product" className="custom-class" />
        )
        expect( wrapper.toJSON() ).toMatchSnapshot();
    });

    test('debe de ostrar el componente con el nombre del producto', () => {
        const wrapper = renderer.create(
            <ProductCard product= { product1 }>
                {
                    () => (
                        <ProductTitle />
                    )
                }

            </ProductCard>
        )
        expect( wrapper.toJSON() ).toMatchSnapshot();;
    });
});
