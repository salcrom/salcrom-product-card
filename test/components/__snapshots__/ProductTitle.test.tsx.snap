// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente correctamente con el t√≠tulo personalizado 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;

exports[`ProductTitle debe de ostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin Imagen
  </span>
</div>
`;
